### ISSUE TITLE
Cache hit doesn't update list

### ISSUE THREAD

**Issue by MJjainam** [mindworker/go-concurrency-solutions#1]
If you look at this line -> https://github.com/loong/go-concurrency-solutions/blob/master/2-race-in-cache/main.go#L43
list is not update after a successful read. 

If we add that then I think we should put a lock around the full Get method.

**Comment by sanmeshkakade** [comment]
I agree with @MJjainam, a cache hit should move it to the front of the linked list which is not happening in the solution as pointed out.
